gzip.go

```diff
@@ -273,6 +328,22 @@ func acceptsGzip(r *http.Request) bool {
 	return acceptedEncodings["gzip"] > 0.0
 }
 
+// returns true if we've been configured to compress the specific content type.
+func handleContentType(contentTypes []string, w http.ResponseWriter) bool {
+	// If contentTypes is empty we handle all content types.
+	if len(contentTypes) == 0 {
+		return true
+	}
+
+	for _, c := range contentTypes {
+		if strings.ToLower(c) == strings.ToLower(w.Header().Get(contentType)) {
```

Comment:
perhaps we should lowercase the contentTypes once when the handler is initialized